### ISSUE TITLE
Tensorflow Lite-GPU dynamic shapes error

### ISSUE THREAD

**Issue by InfiniteLife** [google/automl#540]
Model works on Android CPU (I've tested d0) but it is really slow(around 2-3 fps) so Im trying to run it on Android GPU. The thing is it requires static shapes everywhere inside the network. Relevant issue is here https://github.com/tensorflow/tensorflow/issues/38036 
I thought that problem is in inputs so I tried to convert network with [1,512,512,3] instead of [None,512,512,3] but it still give same error:

`java.lang.RuntimeException: java.lang.IllegalArgumentException: Internal error: Failed to apply delegate: Attempting to use a delegate that only supports static-sized tensors with a graph that has dynamic-sized tensors`.

although it do not prints names of the tensors. Viewing network in Neutron I see that shapes of outputs are not derived
![image](https://user-images.githubusercontent.com/7304884/85393716-1dccd700-b556-11ea-8b40-6002e2d5c465.png)

So Im guessing that problem in the blocks starting with Gather operation.

Is it possible?

And is there any solution to it? I am thinking may be somehow cutting out the block with gather operation from the graph and writing the rest of the operations in Java or C++?

**Comment by fsx950223** [comment]
TF-LITE doesn't support dynamic shape yet.

**Comment by InfiniteLife** [comment]
> TF-LITE doesn't support dynamic shape yet.

yes, so I am thinking on how to make it static. There is example of SSD object detection network with custom layer that wraps the end of the network, but I have no expirience creating such layer. Or second idea is to move dynamic shapes network ending to custom code outside of the network. Anyways Im going to explore both ideas

**Comment by alexiniestags** [comment]
Hello! Any fix for that? I got to convert static the input layer, but the output layer still outputs a dynamic shape related to the number of detections

**Comment by mingxingtan** [comment]
If you just want to pad it to fixed size, maybe you can change this line (set padded=True)?

https://github.com/google/automl/blob/c9faec9d5ebe5678c8f9e255e2d73e36874e3974/efficientdet/keras/postprocess.py#L273

**Comment by alexiniestags** [comment]
Thank you. It has fixed the dynamic shape error. Nevertheless, the GPU delegate api does not support several operations, I'll open a new issue for that. Thank you!

**Comment by nadir121** [comment]
> TF-LITE doesn't support dynamic shape yet.

Man, get outta here.
